### PR TITLE
[DEV-5672] Add status category filter in fetch_jira_issues

### PR DESCRIPTION
**Pull request by gaetano-guerriero**
This change is [Reviewable](https://reviewable.io/reviews/athenianco/athenian-api/3428)

### CODE REVIEW

File: server/athenian/api/internal/miners/jira/issue.py [hunk 1]
```diff
@@ -300,6 +300,7 @@ async def fetch_jira_issues(
     extra_columns: Iterable[InstrumentedAttribute] = (),
     adjust_timestamps_using_prs=True,
     on_raw_fetch_complete: Optional[Callable[[pd.DataFrame], Coroutine]] = None,
+    status_categories: Collection[str] | None = None,
```
Comment (vmarkovtsev) [inline server/athenian/api/internal/miners/jira/issue.py:303]: We should insert it inside `jira_filter`

**Review by vmarkovtsev — CHANGES_REQUESTED** [review]

**Review by gaetano-guerriero — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/athenianco/athenian-api/3428)* status: 0 of 11 files reviewed, 1 unresolved discussion (waiting on @vmarkovtsev)

___
*[`server/athenian/api/internal/miners/jira/issue.py` line 303 at r1](https://reviewable.io/reviews/athenianco/athenian-api/3428#-NPMqibl2C4b2z9hYI3e-r1-303:-NPRoPy4AaaS68PKSDxw:b-896fix) ([raw file](https://github.com/athenianco/athenian-api/blob/88633446ba4c71fb02733a0b37ad68163107c49b/server/athenian/api/internal/miners/jira/issue.py#L303)):*
<details><summary><i>Previously, vmarkovtsev (Vadim Markovtsev) wrote…</i></summary><blockquote>

We should insert it inside `jira_filter`
</blockquote></details>

Done.

**Review by vmarkovtsev — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/athenianco/athenian-api/3428)* status: 0 of 11 files reviewed, all discussions resolved

**Review by vmarkovtsev — APPROVED** [review]
> Reviewed 11 of 11 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/athenianco/athenian-api/3428)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @gaetano-guerriero)

**Pull request merged**